source/common/quic/client_connection_factory_impl.h

```diff
@@ -19,6 +24,9 @@ namespace Quic {
 struct PersistentQuicInfoImpl : public Http::PersistentQuicInfo {
   PersistentQuicInfoImpl(Event::Dispatcher& dispatcher, uint32_t buffer_limit);
 
+  // Get a delegate to access the quic session cache.
+  std::unique_ptr<quic::SessionCache> getQuicSessionCacheDelegate();
```

Comment:
Since this returns a new object, how about "create" instead of "get". Also since it returns a quic::SessionCache not a QuicSessionCacheDelegate, I think I would remove "Delegate" from the name of the method.

Reply:
The return value is a new object indeed, but this method isn't creating a new session cache, but merely a wrapper over existing one. So I'm not sure what would be a more informative/less misleading name. Any suggestion?

Reply:
Since it returns a unique_ptr, I think create() makes the most sense. It create a new Wrapper, but not a new Cache, of course. So how bout if this returns a unique_ptr<SessionCacheWrapper>?